disnake/interactions/base.py

```diff
@@ -649,26 +670,60 @@ def is_done(self) -> bool:
         """
         return self._responded
 
-    async def defer(self, *, ephemeral: bool = False, with_message: bool = False) -> None:
+    @overload
+    async def defer(
+        self,
+        *,
+        with_message: Literal[True],
+        ephemeral: bool = ...,
+    ):
+        ...
+
+    @overload
+    async def defer(
+        self,
+        *,
+        ephemeral: bool = ...,
+    ):
+        ...
+
+    @overload
+    async def defer(
+        self,
+        *,
+        with_message: Literal[False],
+    ):
+        ...
+
+    async def defer(
+        self,
+        *,
+        with_message: bool = MISSING,
+        ephemeral: bool = MISSING,
+    ) -> None:
```

Comment:
this should be a separate PR entirely imo, it seems unrelated to the `suppress_embeds` changes (?)
I'm guessing this was added to support `suppress_embeds` when deferring, but now that it'll likely be added later, these changes should be part of that instead

Reply:
https://github.com/DisnakeDev/disnake/pull/474/commits/345b887fa72de6dd7461ee52041fde8f5e9f240c